### PR TITLE
Introduce Sites

### LINKED ISSUE #2513: Streamline UX
- [x] Rename Gateway Groups to Sites in the UI
- [x] Move Resources under Sites
  - [x] Site -> Gateways -> Resources
- [x] Remove Gateway tags
- [x] Add ability to deploy new Gateway
  - [x] Gateway / client binary to read `FIREZONE_ID` variable
- [x] Replace Resources new form to take site as URL param
- [x] Change the unique constraint on `resources.address` to be unique per `gateway_group_id` too

```
docker run \
  FIREZONE_ID=<Ecto.UUIDGenerate()> \
  FIREZONE_TOKEN=<token> \
  FIREZONE_HOSTNAME=`hostname`
```

### PR DESCRIPTION
**Pull request by AndrewDryga**
Closes #2513

### CODE REVIEW

File: elixir/apps/web/lib/web/live/sites/new.ex [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      <:title :if={not is_nil(@group)}>
+        Deploy your Gateway
+      </:title>
+      <:content>
+        <div class="py-8 px-4 mx-auto max-w-2xl lg:py-16">
+          <.form :if={is_nil(@group)} for={@form} phx-change={:change} phx-submit={:submit}>
+            <div class="grid gap-4 mb-4 sm:grid-cols-1 sm:gap-6 sm:mb-6">
+              <div>
+                <.input
+                  label="Name Prefix"
```
Comment (bmanifold) [inline elixir/apps/web/lib/web/live/sites/new.ex:30]: Should we call this "Site Name" or even just "Name"?

**Review by bmanifold — APPROVED** [review]
Overall, looks good.  Had one small question around naming.

The only other thing that I was wondering is around user experience.  Right now it looks like when you create a site you're immediately taken to the "Deploy Gateway" page and shown a Docker command.  Would it make sense to take the user to the "Show Site" page and then let them choose whether they want to deploy a Gateway or create a Resource first? 
cc: @jamilbk

File: elixir/apps/web/lib/web/live/sites/new.ex (hunk 1 above)
Comment (AndrewDryga) [reply elixir/apps/web/lib/web/live/sites/new.ex:30]: Yeah, I plan to rename them in a separate PR

**Comment by AndrewDryga** [comment]
> The only other thing that I was wondering is around user experience. Right now it looks like when you create a site you're immediately taken to the "Deploy Gateway" page and shown a Docker command. Would it make sense to take the user to the "Show Site" page and then let them choose whether they want to deploy a Gateway or create a Resource first?

We can change it later, but I think it will be a bad UX if we allow having a site without a gateway. And there is no point in creating resources before you deploy anything because you can end up in a situation where you create a resource, install a client, and "nothing works".

**Comment by bmanifold** [comment]
> We can change it later, but I think it will be a bad UX if we allow having a site without a gateway. And there is no point in creating resources before you deploy anything because you can end up in a situation where you create a resource, install a client, and "nothing works".

I understand where you're coming from, but I think I disagree that it's bad UX.  The reason I brought up the change in flow was that I'm nervous it may feel a little jarring to go from setting up a Site to immediately being put in to the Gateway page without any real context.  Would also point out that the UX I described is how one of our closest competitors is designed.  That's not to say we need to always follow what everyone else does, just to say that I feel it's a legitimate UX flow.

**Pull request merged**